source/common/upstream/health_discovery_service.h

```diff
@@ -41,25 +125,45 @@ class HdsDelegate
       std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier>&& message) override;
   void onReceiveTrailingMetadata(Http::HeaderMapPtr&& metadata) override;
   void onRemoteClose(Grpc::Status::GrpcStatus status, const std::string& message) override;
-
+  void sendResponse();
   // TODO(htuch): Make this configurable or some static.
-  const uint32_t RETRY_DELAY_MS = 5000;
+  const uint32_t RetryDelayMilliseconds = 5000;
+  static constexpr uint32_t ClusterConnectionBufferLimitBytes = 12345;
+  static constexpr uint32_t ClusterTimeoutSeconds = 1;
+  uint32_t server_response_ms_ = 1000;
+
+  void
+  processMessage(std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier>&& message);
+
+  void establishNewStream();
+  std::vector<HdsClusterPtr> hdsClusters() { return hds_clusters_; };
 
 private:
   void setRetryTimer();
-  void establishNewStream();
-  void sendHealthCheckRequest();
+  void setServerResponseTimer();
   void handleFailure();
-
   HdsDelegateStats stats_;
   Grpc::AsyncClientPtr async_client_;
   Grpc::AsyncStream* stream_{};
   const Protobuf::MethodDescriptor& service_method_;
   Event::TimerPtr retry_timer_;
-  Event::TimerPtr response_timer_;
   envoy::service::discovery::v2::HealthCheckRequest health_check_request_;
   std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier> health_check_message_;
   std::vector<std::string> clusters_;
+  Runtime::Loader& runtime_;
+  Envoy::Stats::Store& store_stats;
+  Ssl::ContextManager& ssl_context_manager_;
+  Secret::SecretManager& secret_manager_;
+  Runtime::RandomGenerator& random_;
+  Event::Dispatcher& dispatcher_;
+
+  Event::TimerPtr server_response_timer_;
+
+  std::vector<std::vector<Upstream::HealthCheckerSharedPtr>> health_checkers_ptr;
```

Comment:
Nit: this should end with `_`, I would just call it `health_checkers_`. Also, please add a comment explaining what this data structure is and what each dimension does.